Docs refactor (#421)
pydantic/logfire: docs/index.md

# Pydantic Logfire

From the team behind Pydantic, Logfire is a new type of observability platform built on the same belief as our open source library — that the most powerful tools can be easy to use.

Logfire is built on OpenTelemetry, and supports monitoring your application from any language, with particularly great support for Python! [Read more](why-logfire/index.md).

## Getting Started

> This page is a quick walk-through for setting up a Python app:

Reviewer: Do we have somewhere we can direct people who want to see an example that's not Python?

Author: Nope, coming soon

2. Follow the prompts to create your account
3. From your Organisation, click **New project** to create your first project

![Counting size of loaded files screenshot](images/logfire-screenshot-first-steps-first-project.png)

!!! info ""
    The first time you use **Logfire** in a new environment, you'll need to set up a project. A **Logfire** project is a namespace for organizing your data. All data sent to **Logfire** must be associated with a project.

??? tip "You can also create a project via CLI..."
    Check the [SDK CLI documentation](reference/cli.md#create-projects-new) for more information on how to create a project via CLI.

## Install the SDK {#sdk}

1. In the terminal, install the **Logfire** SDK (Software Developer Kit):

{{ install_logfire() }}

2. Once installed, try it out!

```bash
logfire -h
```

3. Next, authenticate your local environment:

```bash
logfire auth
```

!!! info ""
    Upon successful authentication, credentials are stored in `~/.logfire/default.toml`.

## Instrument your project {#instrument}
=== ":material-cog-outline: Development"
    !!! tip "Development setup"

>         During development, we recommend using the CLI to configure Logfire. You can also use a write token.

Reviewer 2: Link to token use docs?

    1. Set your project

    ```bash title="in the terminal:"
    logfire projects use <first-project>
    ```

    !!! info ""
        Run this command from the root directory of your app, e.g. `~/projects/first-project`

    2. Write some basic logs in your Python app

>     ```py title="hello-world.py"

Author: HAHAH I was on a huge tear to rename all the files with hyphens instead of underscores but of course for python file names we want the underscores 😆

    import logfire

    logfire.configure()  # (1)!
    logfire.info('Hello, {name}!', name='world')  # (2)!
    ```

    1. The `configure()` method should be called once before logging to initialize **Logfire**.
    2. This will log `Hello world!` with `info` level.

    !!! info ""
        Other [log levels][logfire.Logfire] are also available to use, including `trace`, `debug`, `notice`, `warn`,
        `error`, and `fatal`.

    3. See your logs in the **Live** view

>     ![Hello world screenshot](images/logfire-screenshot-first-steps-hello-world.png)

Reviewer: @ChristopherGS to update with non-beta screenshot

Reviewer: done

> <!-- ---

Reviewer 3: you don't want HTML quotes around the frontmatter, if you don't want to hide navigation, please remove it.

hide:
- navigation
--- -->

> # Get Started

Reviewer 3: since this is the very start of the index page of our docs, I think we need one paragraph explaining what logfire is and/or selling it a little bit. (a bit like the first two sentences [here](https://docs.pydantic.dev/latest/))

It can be one or two sentences, and I guess should include a link to pydantic.dev/logfire.

Reviewer 3: I also these we need one sentence before the steps start, so overall something like this:

---

# Pydantic Logfire (h1)

From the team behind Pydantic, Logfire is a new type of observability platform built on the same belief as our open source library — that the most powerful tools can be easy to use.

Logfire is built on OpenTelemetry, and supports monitoring your application from any language, but we have particularly great support for Python!

_(Or something)_

## Getting Started (h2)

Below is a quick walk through of how you can start using Logfire with a Python app.

### Logfire Platform (h3)

In the logfire platform:

1. [Log into Logfire](#)
2. Follow the prompts to create your account
3. From your Organisation, click "New project" and create your first project
4. ...

### Install SDK

...

Author: This is done, I linked to the `Why Logfire` section to keep you within the docs, this made the most sense to me.

<!-- - [ ] Create an account
- [ ] Set up your first project
- [ ] Install Logfire SDK
- [ ] Authenticate your local environment
- [ ] Instrument your code

>  -->

Reviewer 3: probably best to just delete this.

Author: Done!

3. From your Organisation, click `New project` and create your first project

![Counting size of loaded files screenshot](images/logfire-screenshot-first-steps-first-project.png)

!!! info
    The first time you use **Logfire** in a new environment, you'll need to set up a project. A **Logfire** project is a namespace for organizing your data. All data sent to **Logfire** must be associated with a project.

??? tip "You can also create a project via CLI..."
    Check the [SDK CLI documentation](reference/cli.md#create-projects-new) for more information on how to create a project via CLI.

## Install SDK {#install}

To install the latest version of the **Logfire** SDK, run:

{{ install_logfire() }}

## Development / Production
=== "Development"
    !!! tip "Development setup"
        During development, we recommend using the CLI to configure Logfire. You can also use a write token.

    ## Authenticate

    Authenticate your local environment with **Logfire** by running:

    ```bash
    logfire auth
    ```

    !!! info
        Upon successful authentication, credentials are stored in `~/.logfire/default.toml`.

    ## Configure your project

>     Point to the project you created by running:

Reviewer 3: ```suggestion
    Tell the logfire SDK which project to send data to, from the root directory of your app e.g. `~/projects/first-project`
```

Author: I added a call-out box to highlight this

    ```bash
    logfire projects use first-project

>     ```

Reviewer 3: ```suggestion
    ```

    INFO: this will store a Write Token which the SDK can use to end data in a file in the current directory at `.logfire/logfire_credentials.json`
```

Author: I added this under an info call-out in a separate commit.

=== "Production"
    !!! tip "Production setup"
        In production, we recommend you provide your write token to the Logfire SDK via environment variables.

    Production tab

## Tracing with Spans

Spans let you add context to your logs and measure code execution time. Multiple spans combine to form a trace,
providing a complete picture of an operation's journey through your system.

```py
from pathlib import Path
import logfire

cwd = Path.cwd()
total_size = 0

> logfire.configure()

Reviewer 3: `logfire.configure()` should come immediately after imports.

Author: This section is on a new page now, fixed there. `/docs/get-started/traces`

with logfire.span('counting size of {cwd=}', cwd=cwd):
    for path in cwd.iterdir():
        if path.is_file():

>             with logfire.span('reading {file}', file=path):

Reviewer 3: ```suggestion
            with logfire.span('reading {path}', path=path.relative_to(cwd)):
```
So the messages aren't so ugly.

Author: Done.

                total_size += len(path.read_bytes())

    logfire.info('total size of {cwd} is {size} bytes', cwd=cwd, size=total_size)
```

In this example:

1. The outer span measures the time to count the total size of files in the current directory (`cwd`).
2. Inner spans measure the time to read each individual file.
3. Finally, the total size is logged.

![Counting size of loaded files screenshot](images/logfire-screenshot-first-steps-load-files.png)

By instrumenting your code with traces and spans, you can see how long operations take, identify bottlenecks,
and get a high-level view of request flows in your system — all invaluable for maintaining the performance and
reliability of your applications.

> [conda]: https://conda.io/projects/conda/en/latest/user-guide/install/index.html

Reviewer 3: This ends a bit abruptly, I think we need a "Next Steps" section, where we say:
* the real power of logfire comes from integrations with many popular libraries, learn more
* as well as spans, you can use logfire to record metrics, learn more
* of course Logfire doesn't just work with Python, learn more

Author: Done 👍
